src/main/java/org/opensearch/sdk/ExtensionsRunner.java

```diff
@@ -393,6 +395,33 @@ public ClusterState sendClusterStateRequest(TransportService transportService) {
         return clusterStateResponseHandler.getClusterState();
     }
 
+    /**
+     * Request the Dependency Information from Opensearch. The result will be handled by a {@Link ExtensionDependencyResponseHandler}.
+     *
+     * @param transportService  The TransportService defining the connection to OpenSearch
+     * @return A List contains details of this extension's dependencies
+     */
+    public List<DiscoveryExtensionNode> sendExtensionDependencyRequest(TransportService transportService) {
+        logger.info("Sending Extension Dependency Information request to Opensearch");
+        ExtensionDependencyResponseHandler extensionDependencyResponseHandler = new ExtensionDependencyResponseHandler();
+        try {
+            transportService.sendRequest(
+                uniqueId,
+                opensearchNode,
+                ExtensionsManager.REQUEST_EXTENSION_DEPENDENCY_INFORMATION,
+                new ExtensionRequest(ExtensionsManager.RequestType.REQUEST_EXTENSION_DEPENDENCY_INFORMATION),
+                extensionDependencyResponseHandler
+            );
+            // Wait on Extension Dependency response
+            extensionDependencyResponseHandler.awaitResponse();
+        } catch (TimeoutException e) {
```

Comment:
After you change `awaitResponse()` you will need to match the caught exceptions here.  Or just catch all exceptions which would include the RunTime `CancellationException`.